From d8a9c3df503f1f53cdaadc8e0b79f6c71dfcc6cc Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 2 Aug 2013 09:25:55 +0000
Subject: [PATCH 0381/1000] systools.lib V1.0.0.7 Mantis 0028323: GPPDev:
 Custom VKP-80II fixed mSysEvent: do not use invalid window handle

git-svn-id: svn://localhost/SelfServiceCommon/trunk@555 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/cpp/Systools/src/mSysEventW32.cpp           | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/SelfServiceCommon/Massai/cpp/Systools/src/mSysEventW32.cpp b/SelfServiceCommon/Massai/cpp/Systools/src/mSysEventW32.cpp
index a4e2d220..e26ec732 100644
--- a/SelfServiceCommon/Massai/cpp/Systools/src/mSysEventW32.cpp
+++ b/SelfServiceCommon/Massai/cpp/Systools/src/mSysEventW32.cpp
@@ -291,7 +291,8 @@ namespace mSysEvent
     m_trc(trc),
     m_threadId(0),
     m_terminate(false),
-    m_deviceEventProcessor(0)
+    m_deviceEventProcessor(0),
+    m_window(0)
   {}
 
   SystemMessageProcessor::~SystemMessageProcessor()
@@ -373,9 +374,12 @@ namespace mSysEvent
 
   void SystemMessageProcessor::postWindowMessage(UINT message, WPARAM wParam, LPARAM lParam)
   {
-    if(!PostMessage(m_window, message, wParam, lParam))
+    if(m_window)
     {
-      mSysError::traceLastError(m_trc, MajorError_HERE, "PostMessage", GetLastError());
+      if(!PostMessage(m_window, message, wParam, lParam))
+      {
+        mSysError::traceLastError(m_trc, MajorError_HERE, "PostMessage", GetLastError());
+      }
     }
   }
 
-- 
2.41.0.windows.1

